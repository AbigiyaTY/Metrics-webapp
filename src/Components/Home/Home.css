.home {
  margin-left: 20px;
}

.topHome {
  background-color: #f0f1f3;
  height: 200px;
  color: #345567;
}

.homeTitle {
  font-size: 20px;
  font-weight: bold;
  text-align: center;
  padding-top: 70px;
}

.homeImg {
  width: 70%;
  padding-top: 70px;
}

.totalStatus {
  background-color: #2a6877;
  color: #000;
}

.titleNews {
  border: 1px solid #000;
}

.titleNews:nth-of-type(n + 1) {
  background-color: #c7d9e5;
  color: #000;
  margin-left: -10px;
}

.titleNews:nth-of-type(n + 3) {
  background-color: #2a6877;
  color: #fff;
}

.titleNews h3 {
  font-size: 14px;
  padding-bottom: 30px;
}

.fa-greater-than {
  padding-left: 104px;
  font-size: 15px;
}

@media screen and (min-width: 600px) {
  .topHome {
    height: 300px;
  }

  .homeTitle {
    font-size: 50px;
    padding-top: 100px;
  }

  .homeImg {
    width: 50%;
    margin-left: 135px;
    padding-top: 57px;
  }

  .fa-greater-than {
    padding-left: 93%;
    font-size: 25px;
  }

  .titleNews h3 {
    font-size: 30px;
    padding-bottom: 49px;
  }
}
