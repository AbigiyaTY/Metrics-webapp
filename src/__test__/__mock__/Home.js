import React from 'react';
import { ListGroup } from 'react-bootstrap';
import { BrowserRouter } from 'react-router-dom';

const Home = () => {
  const news = [{
    id: '1',
    title: 'Tamil Nadu Assembly passes bill to ban online gambling games',
  }];
  return (
    <BrowserRouter>
      <h3>Latest technology news</h3>
      <ListGroup key={news.id}>
        {news.map((i) => (
          <ListGroup.Item key={i.id}>
            <h5>{i.title}</h5>
          </ListGroup.Item>
        ))}
      </ListGroup>
    </BrowserRouter>
  );
};

export default Home;
