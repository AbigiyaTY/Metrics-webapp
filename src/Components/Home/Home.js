import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { fetchNews } from '../../Redux/DetailNews/AllRed';
import HomeImg from '../../Image/News.jpg';
import './Home.css';

const Home = () => {
  const { news } = useSelector((state) => state.news);
  const dispatch = useDispatch();

  useEffect(() => {
    if (news.length === 0) {
      dispatch(fetchNews());
    }
  }, [dispatch, news.length]);

  if (news.length === 0) {
    return (
      <div className="container">
        <div className="row">
          <div className="col-12 row topHome">
            <li className="col-6"><img className="homeImg" src={HomeImg} alt="News" /></li>
            <li className="col-6 homeTitle">
              Latest technology news Loading ...
            </li>
          </div>
        </div>
      </div>
    );
  }

  return (
    <div className="home">
      <div className="container">
        <div className="row">
          <div className="col-12 row topHome">
            <li className="col-6"><img className="homeImg" src={HomeImg} alt="News" /></li>
            <li className="col-6 homeTitle">
              Latest technology news
            </li>
          </div>
          <div className="col-12 row totalStatus">
            <h3 className="col-12 py-2">Total Status</h3>
          </div>
          <div className="col-12 containt">
            <ul className="col-12 row">
              {news.data.map((item) => (
                <li
                  className="col-6 titleNews"
                  key={`${item.id}`}
                >
                  <NavLink
                    state={item}
                    to={`/detail/${item.id}`}
                    key={item.id}
                  >
                    <i
                      className="fa fa-greater-than py-4"
                    />
                  </NavLink>
                  <h3 className="px-2">
                    {item.title}
                  </h3>
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
