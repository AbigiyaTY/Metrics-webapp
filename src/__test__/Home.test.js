import renderer from 'react-test-renderer';
import Home from './__mock__/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
