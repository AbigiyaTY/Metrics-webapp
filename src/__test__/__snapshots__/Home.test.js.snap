// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <h3>
    Latest technology news
  </h3>,
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <h5>
        Tamil Nadu Assembly passes bill to ban online gambling games
      </h5>
    </div>
  </div>,
]
`;
